src/org/labkey/test/tests/component/GridPanelBaseTest.java

```diff
@@ -0,0 +1,134 @@
+    protected static final String VIEW_SAVE = "Save Grid View";
+
+    // Column removed from default view.
+    private static final String REMOVED_FLAG_COLUMN = "Flag";
+
+    @Override
+    protected BrowserType bestBrowser()
+    {
+        return BrowserType.CHROME;
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return "QueryGridBaseTest_Project";
+    }
+
+    @Override
+    public List<String> getAssociatedModules()
+    {
+        return Arrays.asList();
+    }
+
+    // Generate a list of string that has different combinations/sets from the list of strings passed in.
+    // For example if values = ['A', 'B', 'C'] this will return the list ['', 'A', 'B', 'C', 'AB', 'AC', 'ABC', 'BC'].
+    // This is all the sets of the characters from the list, including the empty set.
+    protected List<String> getAllSets(List<String> values, int index)
+    {
+        List<String> allSets = new ArrayList<>();
+        if(index < 0) {
+            allSets.add("");
+            return allSets;
+        }
+
+        allSets = getAllSets(values, index - 1);
+        List<String> newSets = new ArrayList<>();
+
+        for (String allCombination : allSets)
+        {
+            newSets.add(allCombination + values.get(index));
+        }
+
+        allSets.addAll(newSets);
+
+        return allSets;
+    }
+
+    protected void initProject()
+    {
+        _containerHelper.createProject(getProjectName(), null);
+
+        // Add the 'Sample Types' web part. It is easier when debugging etc...
+        PortalHelper portalHelper = new PortalHelper(this);
+        portalHelper.enterAdminMode();
+        portalHelper.addWebPart("Sample Types");
+        portalHelper.exitAdminMode();
+    }
+
+    /**
+     * Helper to remove the 'Flag' column from the default view. It just gets in the way for some tests, and is easier
+     * to remove it.
+     *
+     * @param sampleType Name of sample type.
+     */
+    protected void removeFlagColumnFromDefaultView(String sampleType)
+    {
+        goToProjectHome();
+
+        refresh();
+
+        waitAndClickAndWait(Locator.linkWithText(sampleType));
+
+        log(String.format("Remove '%s' column form default view.", REMOVED_FLAG_COLUMN));
+
+        SampleTypeHelper sampleHelper = new SampleTypeHelper(this);
+        DataRegionTable drtSamples = sampleHelper.getSamplesDataRegionTable();
+        drtSamples.goToView(VIEW_DEFAULT);
+
+        CustomizeView cv = drtSamples.openCustomizeGrid();
+        cv.removeColumn(REMOVED_FLAG_COLUMN);
+
+        // This will save the changes to the default view for everyone. The view menu option will remain as "Default"
+        cv.saveCustomView("", true);
+
+    }
+
+    /**
+     * Make sure there are no filters or search values persisted for the given sample type.
+     *
+     * @param sampleType The sample type used to populate the grid.
+     * @return A queryGrid object.
+     */
+    protected QueryGrid initQueryGrid(String sampleType)
```

Comment:
"init" usually implies configuring the server or test project. Maybe "goto" or "beginAt"?
```suggestion
    protected QueryGrid beginAtQueryGrid(String sampleType)
```

Reply:
Fair enough. beginAt it is.